com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/font/Fontc.java

```diff
@@ -231,17 +244,26 @@ public void TTFBuilder(InputStream fontStream) throws FontFormatException, IOExc
 
         if (!fontDesc.getAllChars()) {
 
-            // 7-bit ASCII. Note inclusive range [32,126]
-            for (int i = 32; i <= 126; ++i) {
-                characters.add(i);
+            String chars = fontDesc.getCharacters();
+            if (!StringUtils.isEmpty(chars)) {
+                for (int i = 0; i < chars.length(); i++) {
+                    char c = chars.charAt(i);
+                    characters.add((int) c);
+                }
             }
-
-            String extraCharacters = fontDesc.getExtraCharacters();
-            for (int i = 0; i < extraCharacters.length(); i++) {
-                char c = extraCharacters.charAt(i);
-                characters.add((int)c);
+            else {
+                for (char c : ASCII_7BIT) {
+                    characters.add((int) c);
+                }
+                String extraCharacters = fontDesc.getExtraCharacters();
+                for (int i = 0; i < extraCharacters.length(); i++) {
+                    char c = extraCharacters.charAt(i);
+                    characters.add((int) c);
+                }
             }
         }
+        Set<Integer> deDup = new HashSet<Integer>(characters);
```

Comment:
Good thinking with the de-duplication!

However, you probably want to use a `SortedSet` here to ensure you get consistent ordering between builds?

Reply:
It makes sense. Thank you! I replaced it with a `TreeSet`